common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -533,14 +533,19 @@ public boolean contentEqualsIgnoreCase(CharSequence string) {
             return false;
         }
 
+        byte[] value = this.value;
         if (string instanceof AsciiString) {
             AsciiString rhs = (AsciiString) string;
-            for (int i = arrayOffset(), j = rhs.arrayOffset(), end = i + length(); i < end; ++i, ++j) {
-                if (!equalsIgnoreCase(value[i], rhs.value[j])) {
-                    return false;
+            byte[] rhsValue = rhs.value;
+            if ((arrayOffset() | rhs.arrayOffset()) == 0) {
```

Comment:
Does it make a different if we are explicit? It can be easier to read.
```suggestion
            if (arrayOffset() == 0 && rhs.arrayOffset() == 0) {
```

Reply:
It could in case it's not the common case that both are true; in few openJDK methods this same pattern is used to reduce the number of branches (and misprediction) for very hot paths.

Said that, IDK if is that frequent that are not both 0, TBH, which instead looks the most common case we care about

Reply:
Sadly, looking at the produced ASM for JDK 17, seems not :/